build/bin/network-validator.go

```diff
@@ -100,17 +105,55 @@ func TestEndpoints(config reachabilityConfig) {
 	os.Exit(0)
 }
 
-func ValidateReachability(host string, port int, tlsDisabled bool) error {
+func embedProxyCertificate(cacertFile string) (*x509.CertPool, error) {
+	// Get the SystemCertPool, continue with an empty pool on error
+	rootCAs, _ := x509.SystemCertPool()
+	if rootCAs == nil {
+		rootCAs = x509.NewCertPool()
+	}
+
+	if cacertFile == "" {
+		return rootCAs, nil
+	}
+
+	// Read in the cert file
+	certs, err := ioutil.ReadFile(cacertFile)
+	if err != nil {
+		return nil, err
+	}
+
+	// Append our cert to the system pool
+	if ok := rootCAs.AppendCertsFromPEM(certs); !ok {
+		log.Println("No certs appended, using system certs only")
+	}
+
+	return rootCAs, nil
+}
+
+func ValidateReachability(host string, port int, tlsDisabled bool, cacertFile string) error {
 	var err error
 	endpoint := fmt.Sprintf("%s:%d", host, port)
 	httpClient := http.Client{
 		Timeout: *timeout,
 	}
 
-	if tlsDisabled {
-		httpClient.Transport = &http.Transport{
-			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
-		}
+	// Setup Certs
+	rootCAs, err := embedProxyCertificate(cacertFile)
```

Comment:
Should this be skipped if none were supplied?

Reply:
yeah.. indeed. I'm moving empty string check to upper context

Reply:
yes, it's now skipped if there's no cert. 
For this scenario, the rootca is being passed as nil and it is harvested as the host's root ca -- aka default

Reply:
yes, I've just updated here: it's now skipped if there's no cert. 
For this scenario, the rootca is being passed as nil and it is harvested as the host's root ca -- aka default